pytorch_pfn_extras/handler.py

```diff
@@ -314,13 +315,15 @@ def train_validation_end(self, trainer, evaluator):
     def _complete_train_step(self, trainer, outs, block, sn, sm, rt):
         idx, batch, cback = self.pending_iters[sn][0]
         self.pending_iters[sn] = self.pending_iters[sn][1:]
+        self.futures[sn] = self.futures[sn][1:]
```

Comment:
Merge this list and objects in a single object with `_PendingAsyncIteration` class?